lib/collections/users.js

```diff
@@ -133,9 +133,66 @@ function checkParentGuardian(age, parentGuardian) {
   }
 }
 
+// Check if the country entered is for the United States.
+function isUnitedStates(country) {
+    let c = country.toUpperCase();
+    return ((c == 'UNITED STATES') ||
+	    (c == 'USA') ||
+	    (c == 'U.S.') ||
+	    (c == 'U.S.A.') ||
+	    (c == 'US'));
+}
+
+// Check if the country entered is for Canada
+function isCanada(country) {
+    let c = country.toUpperCase();
+    return ((c == 'CANADA') || (c == 'CA') || (c == 'CAN'));
+}
+
+// State values are required for the US and Canada; not checked
+// for other countries
+function checkCountryState(data) {
+    let country = data.country;
+    if (isUnitedStates(country) || isCanada(country)) {
+	if ((data.state == '') || (data.state == 'OTHER')) {
+	    throw new Meteor.Error(400, 'State/province required for US and Canadian addresses');
+	}
+    }
+}
+
+// If this is a US address, check that the zip code conforms to a zip
+// or zip+4 pattern. If this is a Canadian address, check that the
+// postcode conforms to the Canadian code pattern. Do not check other
+// countries.
+function checkPostalCode(country, postcode) {
+    let okay = true;
+    if (isUnitedStates(country)) {
+	if (postcode.length == 0) {
+	    throw new Meteor.Error(400, 'Zip code required for US addresses');
+	}
+	const zip5 = /^\d{5}$/;
+	const zipplus4 = /^\d{5}\-\d{4}/;
+	if (!(postcode.match(zip5) || postcode.match(zipplus4))) {
+	    throw new Meteor.Error(400, 'Invalid Zip code format for a US address'); 
+	}
+    } else if (isCanada(country)) {
+	if (postcode.length == 0) {
+	    throw new Meteor.Error(400, 'Postal code required for Canadian addresses');
+	}
+	let pc = postcode.toUpperCase();
+	const pcspace = /^[A-Z]\d[A-Z]\s\d[A-Z]\d$/;
+	const pcnospace = /^[A-Z]\d[A-Z]\d[A-Z]\d$/;
+	if (!(pc.match(pcspace) || pc.match(pcnospace))) {
+	    throw new Meteor.Error(400, 'Invalid postal code format for a Canadian address'); 
+	}
+    }
+}
+
 // User collections methods:
 Meteor.methods({
-  'user.register'(data) {
+    'user.register'(data) {
+	console.log('in user.register, data is:');
+	console.log(data);
```

Comment:
Did you mean to include these lines?

Reply:
I did not. Glad you caught this. Removed.